server/tools/x509.sh

```diff
@@ -16,21 +16,33 @@ function autogenerate_keystores() {
     local X509_KEYSTORE_DIR="/etc/x509/${KEYSTORE_TYPE}"
     local X509_CRT="tls.crt"
     local X509_KEY="tls.key"
+    # read X509_KEY_PASS environment variable if set or use default value: none
+    local X509_KEY_PASSWORD=${X509_KEY_PASS:="none"}
     local NAME="keycloak-${KEYSTORE_TYPE}-key"
     local PASSWORD=$(openssl rand -base64 32 2>/dev/null)
     local JKS_KEYSTORE_FILE="${KEYSTORE_TYPE}-keystore.jks"
     local PKCS12_KEYSTORE_FILE="${KEYSTORE_TYPE}-keystore.pk12"
 
     if [ -f "${X509_KEYSTORE_DIR}/${X509_KEY}" ] && [ -f "${X509_KEYSTORE_DIR}/${X509_CRT}" ]; then
 
-      echo "Creating ${KEYSTORES[$KEYSTORE_TYPE]} keystore via OpenShift's service serving x509 certificate secrets.."
-
-      openssl pkcs12 -export \
-      -name "${NAME}" \
-      -inkey "${X509_KEYSTORE_DIR}/${X509_KEY}" \
-      -in "${X509_KEYSTORE_DIR}/${X509_CRT}" \
-      -out "${KEYSTORES_STORAGE}/${PKCS12_KEYSTORE_FILE}" \
-      -password pass:"${PASSWORD}" >& /dev/null
+        echo "Creating ${KEYSTORES[$KEYSTORE_TYPE]} keystore via OpenShift's service serving x509 certificate secrets.."
```

Comment:
- I will create a keytools.sh script that pre-processes the certificates and keys so that the X509.sh deals exclusively with autogenerating keystores.
- I will modify the docker-entrypoint.sh to execute the keytools.sh before executing X509.sh
- I will take care of commits once done 

cheers

Reply:
Awesome! Thank you!